agent/main.go

```diff
@@ -349,3 +388,199 @@ func mainImpl() {
 		os.Exit(1)
 	}
 }
+
+// Watch for CM creation.
+func watchConfigMaps(cfg *agentConfig) {
+	source := cache.NewListWatchFromClient(
+		cfg.KubeClient.Core().RESTClient(),
+		"configmaps",
+		"weave",
+		fields.SelectorFromSet(fields.Set{"metadata.name": "cloudwatch"}))
+
+	cfg.CMInformer = cache.NewSharedIndexInformer(
+		source,
+		&apiv1.ConfigMap{},
+		0,
+		cache.Indexers{},
+	)
+
+	cfg.CMInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    cfg.handleCMAdd,
+		UpdateFunc: cfg.handleCMUpdate,
+		DeleteFunc: cfg.handleCMDelete,
+	})
+}
+
+// Triggered on all ConfigMap creation with name cloudwatch in weave ns
+func (cfg *agentConfig) handleCMAdd(obj interface{}) {
+	cm, ok := obj.(*apiv1.ConfigMap)
+	if !ok {
+		log.Error("Failed to type assert ConfigMap")
+		return
+	}
+
+	cfg.conform(cm)
+}
+
+// Triggered on all ConfigMap update with name cloudwatch in weave ns
+func (cfg *agentConfig) handleCMUpdate(old, cur interface{}) {
+	log.Info("ConfigMap was updated")
+	log.Infof("old: %v cur: %v", old, cur)
+	cm, ok := cur.(*apiv1.ConfigMap)
+	if !ok {
+		log.Error("Failed to type assert ConfigMap")
+		return
+	}
+
+	cfg.conform(cm)
+}
+
+// Triggered on only ConfigMap deletion with name cloudwatch in weave ns
+func (cfg *agentConfig) handleCMDelete(obj interface{}) {
+	log.Info("ConfigMap was deleted")
+	// TODO: Handle ConfigMap deletion. Delete individual objects that were created as at this point.
+	log.Infof("obj: %v", obj)
+}
+
+// Watch for Secret creation/update/deletion.
+func watchSecrets(cfg *agentConfig) {
+	source := cache.NewListWatchFromClient(
+		cfg.KubeClient.Core().RESTClient(),
+		"secrets",
+		"weave",
+		fields.Everything())
+
+	cfg.SecretInformer = cache.NewSharedIndexInformer(
+		source,
+		&apiv1.Secret{},
+		0,
+		cache.Indexers{},
+	)
+
+	cfg.SecretInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    cfg.handleSecretAdd,
+		UpdateFunc: cfg.handleSecretUpdate,
+		DeleteFunc: cfg.handleSecretDelete,
+	})
+}
+
+// Triggered on all Secret created in the weave ns
+func (cfg *agentConfig) handleSecretAdd(obj interface{}) {
+	_, ok := obj.(*apiv1.Secret)
+	if !ok {
+		// If the object is not a secret we should ignore.
+		log.Error("Failed to type assert Secret")
+		return
+	}
+
+	if err := cfg.conformSecret(); err != nil {
+		log.Error(err)
+		return
+	}
+}
+
+// Triggered on all Secret updates in the weave ns
+func (cfg *agentConfig) handleSecretUpdate(old, cur interface{}) {
+	_, ok := cur.(*apiv1.Secret)
+	if !ok {
+		// If the object is not a secret there was an error.
+		log.Error("Failed to type assert Secret")
```

Comment:
Nit: `!ok` seems unlikely to happen, but if it does, then would it be worth logging the type of the object?